src/DynamoCore/Updates/UpdateManager.cs

```diff
@@ -64,6 +64,13 @@ public interface IDynamoLookUp
         IEnumerable<string> GetDynamoInstallLocations();
 
         /// <summary>
+        /// Gets the full path of user data location of all version of this
+        /// Dynamo product installed on this system.
+        /// </summary>
+        /// <returns></returns>
+        IEnumerable<string> GetDynamoUserDataLocations();
```

Comment:
/// <summary>
/// Get a list of user data folders on this system.
/// </summary>
/// <returns>
/// The implementation of this interface method should return a list of user 
/// data folders, one for each of Dynamo product installed on the system. When 
/// there is no Dynamo product installed, this method returns an empty list.
/// </returns>